airstack-modules/modules/airstack/domain-name/domain-name.ts

```diff
@@ -180,178 +144,153 @@ export namespace domain {
 
   /**
    * @dev This function tracks a new resolver transaction
-   * @param resolver resolver contract address
-   * @param node domain node
-   * @param chainId chain id
-   * @param blockHeight block number in the chain
-   * @param blockHash block hash
-   * @param blockTimestamp block timestamp
+   * @param block ethereum block
    * @param transactionHash transaction hash
-   * @param logIndex event log index
+   * @param logOrCallIndex txn log or call index
+   * @param domainId air domain id
+   * @param resolver resolver contract address
    * @param tokenAddress contract address of nft token
-   * @param fromOldRegistry specifies if the event is from the old registry
    */
   export function trackDomainNewResolverTransaction(
-    resolver: string,
-    node: string,
-    chainId: string,
-    blockHeight: BigInt,
-    blockHash: string,
-    blockTimestamp: BigInt,
+    block: ethereum.Block,
     transactionHash: string,
-    logIndex: BigInt,
+    logOrCallIndex: BigInt,
+    domainId: string,
+    resolver: string,
     tokenAddress: string,
-    fromOldRegistry: boolean,
   ): void {
+    let chainId = processChainId();
     // get block
-    let block = getOrCreateAirBlock(chainId, blockHeight, blockHash, blockTimestamp);
+    let airBlock = getOrCreateAirBlock(chainId, block.number, block.hash.toHexString(), block.timestamp);
     // get domain
-    let domain = getOrCreateAirDomain(new Domain(node, chainId, block, tokenAddress));
-    let isMigratedMapping = getOrCreateIsMigratedMapping(domain.id, block.id);
-    if (fromOldRegistry && node != ROOT_NODE && isMigratedMapping.isMigrated == true) {
-      // this domain was migrated from the old registry, so we don't need to hanlde old registry event now
-      return;
-    }
+    let domain = getOrCreateAirDomain(new Domain(domainId, chainId, airBlock, tokenAddress));
     // get previous resolver
     let previousResolverId = domain.resolver;
     // create new resolver
-    let resolverEntity = getOrCreateAirResolver(domain, chainId, resolver);
+    let resolverEntity = getOrCreateAirResolver(domain, chainId, airBlock, resolver);
```

Comment:
are you using airBlock is resolver entity?

Reply:
air block is being used to create air account, and not being saved inside resolver